### PR TITLE
Add media gallery block

### PR DESCRIPTION
**Pull request by vladciocoiu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Partial fix for #5 #6 

Test URLs:
- Original: https://www.esri.com/en-us/about/about-esri/overview
- Before: https://main--esri--aemsites.hlx.live/drafts/shared/media-gallery
- After: https://add-media-gallery-block--esri--vladciocoiu.hlx.live/drafts/shared/media-gallery

### CODE REVIEW

**Comment by alexcarol** [comment]
For #6 (in https://www.esri.com/en-us/capabilities/imagery-remote-sensing/capabilities/analysis) the whole card becomes a link. I'd suggest adding logic to wrap the card in a link for this task *and then* for the follow-up task you can rewrite the link to open the lightbox (this is most likely something we want to do across the site, as it affects other links to media, not just for cards).

File: blocks/media-gallery/media-gallery.css [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    width: 100vw;
+}
+.media-gallery {
+    display: grid;
+    grid-gap: 1rem;
+    width: calc(100vw - 4rem);
+    margin: auto
+}
+
+.media-gallery p {
```
Comment (alexcarol) [inline blocks/media-gallery/media-gallery.css:11]: I'd suggest using css nesting, let's let @tim-white-esri confirm that's fine with him (support is pretty widespread already: https://caniuse.com/css-nesting)

File: blocks/media-gallery/media-gallery.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        child.classList.add('card');
+        child.classList.add(`card-${idx}`);
+
+        const cardContent = document.createElement('div');
+        cardContent.classList.add('card-content');
+
+        console.log(block.classList);
+        if(block.classList.contains('alternate-2-1')) {
+            const startButton = document.createElement('button');
+            startButton.classList.add('start-button');
```
Comment (alexcarol) [inline blocks/media-gallery/media-gallery.js:16]: I wouldn't make this button dependent on the "alternate-2-1", but based on detection of a link to a video (for now i'd use https://mediaspace.esri.com/, maybe we can get confirmation as to whether everything there are videos or we need to add a explicit variant to the block).

File: blocks/media-gallery/media-gallery.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            startButton.innerHTML = "<svg aria-hidden='true' class='svg' fill='currentColor' height='100%' viewBox='0 0 24 24' width='100%' xmlns='http://www.w3.org/2000/svg'><path d='M6 1.773l15 10.23L6 22.226z'></path></svg>";
```
Comment (alexcarol) [inline blocks/media-gallery/media-gallery.js:17]: As they're using a "calcite-button", can you use that directly instead of a custom implementation? Their calcite web components should be available to use within the project already.
```
<calcite-button appearance="solid" kind="inverse" color="light" scale="l" round="" icon-start="play-f" dir="ltr" alignment="center" width="auto" label="play video - How does the US count everyone?" calcite-hydrated="" type="button">
	</calcite-button>
```

Comment (alexcarol) [reply blocks/media-gallery/media-gallery.js:17]: I'd also suggest using dom-helpers to add the components and adding calcite component helpers for it as needed (calciteButton, calciteCard, ...).
https://github.com/andreituicu/franklin-jsx-like/blob/main/scripts/dom-helpers.js

This is script is becoming a standard in most Edge Delivery projects.

File: blocks/media-gallery/media-gallery.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        const cardTitle = child.children[0].children[0];
+        const cardDescription = child.children[0].children[1];
+
+        cardTitle.classList.add('card-title');
+        cardDescription.classList.add('card-description');
+
+        cardContent.appendChild(cardTitle);
+        cardContent.appendChild(cardDescription);
+
+        child.children[0].appendChild(cardContent);
```
Comment (alexcarol) [inline blocks/media-gallery/media-gallery.js:30]: you're using child.children[0] in many places, can you move it to a variable for readability?

File: styles/styles.css [hunk 5]
```diff
@@ -20,7 +20,21 @@
   --text-color: black;
 
   /* fonts */
-  --body-font-family: "Avenir Next";
```
Comment (alexcarol) [inline styles/styles.css:23]: are any of the changes in this file intended?

Comment (vladciocoiu) [reply styles/styles.css:23]: I guess yes, I found some more general font sizes and families on the website, and I tought they would be helpful

Comment (alexcarol) [reply styles/styles.css:23]: But why did you change this variable "back" to roboto?

File: styles/styles.css [hunk 6]
```diff
@@ -185,7 +200,19 @@ main img {
 
 /* sections */
 main .section {
-  padding: 64px 16px;
+  padding: 64px 0;
+}
+
+@media (width >= 600px) {
+  main .section {
+    padding: 64px 0;
```
Comment (alexcarol) [inline styles/styles.css:208]: Why is this needed?

File: styles/styles.css (hunk 5 above)
Comment (vladciocoiu) [reply styles/styles.css:23]: Oops, my mistake. Fixed it now

File: styles/styles.css (hunk 6 above)
Comment (vladciocoiu) [reply styles/styles.css:208]: I think I changed it because it got in the way of the margin the media gallery needed to have. I could just use this for the space needed, if you think it is better this way, but it needs to be adjusted regardless

File: blocks/media-gallery/media-gallery.css [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    .cards-card-body:is(:hover,:focus,:focus-within) img {
+        transform: scale(1.08);
+    }
+    .start-button {
+        margin: 0;
+        margin-bottom: 8px;
+        width: 44px;
+        height: 44px;
+        color: rgb(21, 21, 21);
+        background-color: rgb(248, 248, 248);
```
Comment (alexcarol) [inline blocks/media-gallery/media-gallery.css:97]: Can you use hex codes for the color codes? (mostly for uniformity)

File: styles/styles.css (hunk 6 above)
Comment (alexcarol) [reply styles/styles.css:208]: I mean that it looks like it should not be needed because it has the same exact value as the default right above

**Review by alexcarol — APPROVED** [review]

File: blocks/media-gallery/media-gallery.css (hunk 1 above)
Comment (tim-white-esri) [reply blocks/media-gallery/media-gallery.css]: Fine with me. I haven't used native CSS nesting yet; looking forward to trying it out.

File: blocks/media-gallery/media-gallery.css (hunk 7 above)
Comment (tim-white-esri) [reply blocks/media-gallery/media-gallery.css]: Ideally we should use the Calcite color variables. They have our approved hex colors and work in light/dark modes. 
 --calcite-ui-brand
    --calcite-ui-brand-hover
    --calcite-ui-brand-press
    --calcite-ui-background-1
    --calcite-ui-foreground-1
    --calcite-ui-foreground-2
    --calcite-ui-foreground-3
    --calcite-ui-text-1
    --calcite-ui-text-2
    --calcite-ui-text-3
    --calcite-ui-text-inverse
    --calcite-ui-text-link
    --calcite-ui-border-1
    --calcite-ui-border-2
    --calcite-ui-border-3
